### DISCUSSION TITLE
Allow for generic webhook format

### DISCUSSION THREAD

**Discussion by mew1033** [mbround18/valheim-docker#767] [Q&A]
I'd like to use pushover for my notifications, not discord. I've read the docs, but I can't tell if there's a way to do this or not. I need to follow this api: https://pushover.net/api

Thanks!

**Comment by spannerman79** [comment]
**If** this is to be added I suggesting splitting the webook section up.

For example;
`WEBHOOK_ENABLE` defaults to `false`
`WEBHOOK_DISCORD`, defaults to `false`, when `true` everything currently in https://github.com/mbround18/valheim-docker/blob/main/docs/webhooks.md#environment-variables is active/able to be configured.

Just change `WEBHOOK_BROADCAST_MESSAGE` to `WEBHOOK_DISCORD_BROADCAST_MESSAGE` and so on for the specific discord stuff.

For any other add on webhooks that might be added on in the future any en vars prob start with `WEBHOOK_{name}` to keep things some what sane.

Unless someone else have some other better suggestion to organize something like this

**Comment by mbround18** [comment]
@mew1033 can you give me an example of how the body of the message would look? This can be integrated via url identification and body message adaptation. We already support generic webhook support, if your url is not a discord webhook url it treats it as a generic and only sends the needed data.

**Reply by mew1033** [reply]
Sure thing. So to send to pushover, you POST to https://api.pushover.net/1/messages.json with a body like so:
```json
{
    "token": "APP_TOKEN",
    "user": "USER_KEY",
    "message": "hello world"
}
```
That's the bare minimum, but you can add a ton more parameters, all listed here: https://pushover.net/api


My end goal is to recreate something I was using with the [other valheim docker server](https://github.com/lloesche/valheim-server-docker) I was running. I want to get pushover notifications when the server starts and when people join. On that container, I was using `PRE_BOOTSTRAP_HOOK` and `VALHEIM_LOG_FILTER_CONTAINS_Spawned`. I was just using `curl` to hit the pushover api directly.

**Comment by mew1033** [comment]
Another (IMO awesome) option would be to integrate https://github.com/caronc/apprise. That way you could just build support for a single service and people could send notifications to whatever system they want.
